### PR TITLE
uVision Exporter - Add support for ASM Macros

### PR DESCRIPTION
**Pull request by fvincenzo**

### CODE REVIEW

**Comment by 0xc0170** [comment]
I hope I'll find time this weekend to look at this PR, why it fails for python 3 and why we need this.

**Comment by 0xc0170** [comment]
I am locally having python 3.5.1, and it passes those tests. I'll have to install 3.3/3.4

**Comment by 0xc0170** [comment]
@fvincenzo looking at the code, I believe the line 228 in uvision.py:

```
uvproj_dic[self.FLAGS_TO_UVISION[misc_keys]]['VariousControls']['MiscControls']
```

We haven't encoded those misc keys, you only did it for asm

**Comment by fvincenzo** [comment]
@0xc0170 I will try and let you know.

File: project_generator/tools/uvision.py [hunk 1]
```diff
@@ -222,7 +225,7 @@ def _uvproj_set_TargetArmAds(self, uvproj_dic, project_dic):
                     uvproj_dic[self.FLAGS_TO_UVISION[misc_keys]]['Misc'] += ' ' + item
             else:
                 for item in project_dic['misc'][misc_keys]:
-                    uvproj_dic[self.FLAGS_TO_UVISION[misc_keys]]['VariousControls']['MiscControls'] += ' ' + item
+                    uvproj_dic[self.FLAGS_TO_UVISION[misc_keys.encode('utf-8')]]['VariousControls']['MiscControls'] += ' '.encode('utf-8') + item.encode('utf-8')
```
Comment (0xc0170) [inline project_generator/tools/uvision.py]: ``misc_keys.encode('utf-8')`` not needed as a key in the dictionary ;) Otherwise it makes a key b`name`

**Pull request closed without merging**